pkg/encrypt/interface.go

```diff
@@ -14,31 +14,37 @@
  * limitations under the License.
  */
 
-package minio
+// Package encrypt implements a generic interface to encrypt any stream of data.
+// currently this package implements two types of encryption
+// - Symmetric encryption using AES.
+// - Asymmetric encrytion using RSA.
+package encrypt
 
-import (
-	"io"
-	"net/http"
-)
+import "io"
 
-// EncryptionMaterials - provides generic interface to encrypt
-// any stream of data. Some crypt information can be
-// save in the object metadata
-type EncryptionMaterials interface {
+// Materials - provides generic interface to encrypt any stream of data.
+type Materials interface {
```

Comment:
An IV and Key are much to do with Crypto rather than to do with AWS, AWS SPEC just names them properly and uses them. 

So it is correct the way it is right now, perhaps i can change it.. further.